Support ability to match by regex

lambdachad opened the pull request:
Hey @neg-serg, saw you made some cool changes to this repository. Are the regex changes backwards-compatible with the current code? If so, I don't mind merging it so we can support more stuff.

neg-serg committed:
Add flexible matching options

neg-serg committed:
Update README.md

neg-serg committed:
Add matcher unit tests

neg-serg committed:
feat: add tag/xdgtag shorthands and Hyprland tags array support

- Add --tag and --xdgtag args that expand to matchers
- Match Tag against legacy tag and modern tags array
- Update usage and examples in README

neg-serg committed:
nix: update flake inputs (nixpkgs/naersk) to support Cargo.lock v4 and build successfully

neg-serg commented:
Hi! Yes, the regex changes are mostly backwards-compatible with the current code. The only thing that isn’t working yet is tag support — I was planning to fix that a bit later.

Let me know if you want me to prioritize that.

The reason for this change is that it allows you to write expressions like this:

```
class:^(firefox|org\.mozilla\.firefox|librewolf|io\.gitlab\.librewolf-community|floorp|one\.ablaze\.floorp|floorpdeveloperedition|yandex-browser|Yandex-browser|chromium|org\.chromium\.Chromium|thorium-browser|com\.thorium\.Thorium|palemoon|net\.palemoon\.PaleMoon)$
```

As a result, this makes it possible to cycle through all applications that meet the condition

svelterust commented:
Yes, when you consider it ready I'll take a look then we can get it merged. As long as the current code and the examples in this repository works as is, it'd be great to get this merged too.

neg-serg committed:
fix: use sh -c instead of hyprctl keyword exec for launching

hyprctl keyword exec is broken under Hyprland Lua config (v0.55+).
Use direct shell execution instead, matching how hl.dsp.exec_cmd works.

neg-serg committed:
Use Lua-compatible dispatch syntax for Hyprland 0.55.4

hyprctl dispatch focuswindow address:... is broken in Hyprland 0.55.4
with Lua config manager. Use hl.dsp.focus({window = ...}) instead.
